presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -197,16 +215,25 @@ private boolean canAccessCatalog(Identity identity, String catalogName)
     @Override
     public void checkCanCreateSchema(SystemSecurityContext context, CatalogSchemaName schema)
     {
+        if (!canAccessCatalog(context.getIdentity(), schema.getCatalogName(), AccessMode.ALL)) {
+            denyCreateSchema(schema.toString());
```

Comment:
This isn't related to the fact you're introducing "read-only' access mode, so this should be as a separate commit (may be in the same PR).

eg.
- first commit introduces read-only mode and all the related changes but does not add checks to `checkCanCreateSchema, checkCanDropSchema, checkCanRenameSchema` methods
- second commit (`Add schema checks in file-based SystemAccessControl`) fills these methods